listenbrainz/stats/user_similarity.py

```diff
@@ -0,0 +1,48 @@
+import json
+from listenbrainz.db.stats import get_artists_for_all_users
+import listenbrainz.db.user as db_user
+from scipy.sparse import csr_matrix
+import implicit
+
+
+def get_artist_user_matrix(user_artist_stats):
+    """ Returns a Scipy sparse matrix with artists as rows, users as column
+    and the listen count as the value. This is then used to train the CF model.
+    """
+    artist_ids = {}
+    count = 0
+    data = []
+    row_ind = []
+    col_ind = []
+    for stat_record in user_artist_stats:
+        for artist in stat_record.all_time.artists:
+            if artist.artist_name in artist_ids: #TODO: use artist MSID or MBID
```

Comment:
I think running the alg on artists_credit_ids is best and then translating them to MBIDs as a post processing step.